### PR TITLE
Fe/uint256 refactor

### PR DESCRIPTION
**Pull request by pablodeymo**

### CODE REVIEW

File: eth_verifier/lib/bn254/BN254.sol [hunk 1]
```diff
@@ -146,7 +146,7 @@ library BN254 {
 
     /// @return r the product of a point on G1 and a scalar, i.e.
     /// p == p.mul(1) and p.add(p) == p.mul(2) for all points p.
-    function scalarMul(G1Point memory p, uint256 s) internal view returns (G1Point memory r) {
+    function scalarMul(G1Point memory p, uint256 s) public view returns (G1Point memory r) {
```
Comment (gabrielbosio) [inline eth_verifier/lib/bn254/BN254.sol:149]: Try changing function visibility back to `internal` to see if we can lower the gas consumption.

File: eth_verifier/lib/bn254/Fields.sol [hunk 2]
```diff
@@ -89,61 +87,59 @@ library Scalar {
             // data
             o := mload(p)
         }
-        result = FE.wrap(o);
+        result = o;
```
Comment (gabrielbosio) [inline eth_verifier/lib/bn254/Fields.sol:90]: We can remove the `o` variable and directly using `result`.

File: eth_verifier/lib/bn254/Fields.sol [hunk 3]
```diff
@@ -89,61 +87,59 @@ library Scalar {
...
-    function zero() internal pure returns (FE) {
-        return FE.wrap(0);
+    function zero() internal pure returns (uint256) {
+        return 0;
     }
 
-    function one() internal pure returns (FE) {
-        return FE.wrap(1);
+    function one() internal pure returns (uint256) {
+        return 1;
```
Comment (gabrielbosio) [inline eth_verifier/lib/bn254/Fields.sol:103]: Are these functions necessary?

File: eth_verifier/test/Fields.t.sol [hunk 4]
```diff
@@ -8,40 +8,35 @@ import "../lib/bn254/BN256G2.sol";
...
-using {Base.add, Base.mul} for Base.FE;
-using {Scalar.add, Scalar.mul, Scalar.inv} for Scalar.FE;
 using {Oracles.to_field} for Oracles.ScalarChallenge;
 
 contract FieldsTest is Test {
     function test_add_base() public {
-        Base.FE q = Base.FE.wrap(Base.MODULUS - 1);
-        Base.FE one = Base.FE.wrap(1);
+        uint256 q = Base.MODULUS - 1;
+        uint256 one = 1;
```
Comment (gabrielbosio) [inline eth_verifier/test/Fields.t.sol:16]: We can remove this variable and replace it where used.

File: eth_verifier/test/Fields.t.sol [hunk 5]
```diff
@@ -8,40 +8,35 @@ import "../lib/bn254/BN256G2.sol";
...
-        assertEq(Base.FE.unwrap(q_plus_one), 0, "p != 0 mod p");
+        uint256 q_plus_one = Base.add(q, one);
+        assertEq(q_plus_one, 0, "p != 0 mod p");
     }
 
     function test_add_scalar() public {
-        Scalar.FE q = Scalar.FE.wrap(Scalar.MODULUS - 1);
-        Scalar.FE one = Scalar.FE.wrap(1);
+        uint256 q = Scalar.MODULUS - 1;
+        uint256 one = 1;
```
Comment (gabrielbosio) [inline eth_verifier/test/Fields.t.sol:24]: We can remove this variable and replace it where used.

**Comment by gabrielbosio** [comment]
In another PR, we can change how we compute chained operations over fields. Instead of doing for example `sub(x, mul(a, b))` we can do `from(x - a * b)`. This way we reduce the amount of mod operations.

File: eth_verifier/lib/bn254/Fields.sol (hunk 2 above)
Comment (pablodeymo) [reply eth_verifier/lib/bn254/Fields.sol:90]: done! thanks!!!!

File: eth_verifier/lib/bn254/Fields.sol (hunk 3 above)
Comment (pablodeymo) [reply eth_verifier/lib/bn254/Fields.sol]: done!

File: eth_verifier/lib/bn254/BN254.sol (hunk 1 above)
Comment (pablodeymo) [reply eth_verifier/lib/bn254/BN254.sol]: done

File: eth_verifier/test/Fields.t.sol (hunk 4 above)
Comment (pablodeymo) [reply eth_verifier/test/Fields.t.sol:16]: done

**Review by gabrielbosio — APPROVED** [review]

**Pull request merged**